xdsl/frontend/jaxpr/primitive_builder.py

```diff
@@ -0,0 +1,33 @@
+from collections.abc import Callable
+from typing import ClassVar
+
+from jax._src.core import JaxprEqn
+
+from xdsl.dialects.stablehlo import AddOp
+from xdsl.ir import Operation, SSAValue
+
+
+class PrimitiveBuilder:
+    BUILDER_BY_PRIMITIVE: ClassVar[
+        dict[str, Callable[[JaxprEqn, tuple[SSAValue, ...]], Operation]]
```

Comment:
as “add” is not primitive in Jax, I changed `dict[jax_primitive, …. ]` to `dict[str, …. ]`

Reply:
It would be great to try and find exactly what Jax uses for this. Strings are quite dangerous, because `add` is so generic, what if someone comes and calls their expression `add` too? Ideally we'd want to go from some notion of type or function identity. I'm happy to help with this

Reply:
in Jax add_p is a primitive, add is just a wrapper function around it. Using lax.add_p I can still use dict[jax_primitive, …. ]. :)